packit_service/worker/handlers/copr.py

```diff
@@ -479,3 +496,157 @@ def handle_testing_farm(self):
                         "build_id": self.build.id,
                     },
                 ).apply_async()
+
+    def handle_scan(self):
+        """
+        If configured, try to find a job that can provide the base SRPM,
+        download both SRPM and base SRPM and trigger the scan in OpenScanHub.
+        """
+        if (
+            self.db_project_event.type != ProjectEventModelType.pull_request
+            or self.build.target != "fedora-rawhide-x86_64"
+            or not self.job_config.osh_diff_scan_after_copr_build
+        ):
+            return
+
+        base_build_job = self.find_base_build_job()
+        if not base_build_job:
+            logger.debug("No base build job needed for diff scan found in the config.")
+            return
+
+        logger.info("Preparing to trigger scan in OpenScanHub...")
+
+        base_srpm_model = self.get_base_srpm_model(base_build_job)
+
+        if not base_srpm_model:
+            return
+
+        srpm_model = self.build.get_srpm_build()
+
+        directory = tempfile.mkdtemp()
+        paths = self.download_srpms(directory, base_srpm_model, srpm_model)
+        if not paths:
+            return
+
+        build_dashboard_url = get_copr_build_info_url(self.build.id)
+
+        output = self.copr_build_helper.api.run_osh_build(
+            srpm_path=paths[1],
+            base_srpm=paths[0],
+            comment=f"Submitted via Packit Service for {build_dashboard_url}.",
+        )
+
+        if not output:
+            logger.debug("Something went wrong, skipping the reporting.")
+            return
+
+        logger.info(f"Scan submitted successfully, output: {output}")
+
+        response = json.loads(output)
+        url = response.get("url")
+
+        if not url:
+            logger.debug("It was not possible to get the URL from the response.")
+            return
+
+        self.copr_build_helper._report(
+            state=BaseCommitStatus.success,
+            description=(
+                "Scan in OpenScanHub submitted succesfully. Check the URL for more details."
+            ),
+            url=url,
+            check_names=["osh-diff-scan:fedora-rawhide-x86_64"],
```

Comment:
Do we need `fedora-rawhide-x86_64` if we run it just for rawhide? or are we planning to run it for different releases/architectures too? IMO the SRPM should be the same, shouldn't it?

Reply:
The SRPM is here just to get the info on how the project is built so yes, this shouldn't be version-dependent.

Reply:
yes this is true about the SRPMs, but we specify `fedora-rawhide` as a mock config name when submitting the scan as well, so even if run it for just `rawhide`, for explicitness and consistency I wanted rather to include it

Reply:
fine with me